src/sql/connection.py

```diff
@@ -193,11 +196,22 @@ def _error_module_not_found(cls, e):
         return ModuleNotFoundError("test")
 
     def __init__(self, engine, alias=None):
-        self.dialect = engine.url.get_dialect()
-        self.metadata = sqlalchemy.MetaData(bind=engine)
+        self.url = engine.url
         self.name = self.assign_name(engine)
+        self.dialect = self.url.get_dialect()
+        if IS_SQLALCHEMY_ONE < 2:
```

Comment:
this is a bool since we're making the comparison already

```suggestion
        if IS_SQLALCHEMY_ONE:
```